src/server.mjs

```diff
@@ -1154,13 +1303,27 @@ function getBedRegions(bedFile, dataPath) {
     }
     bed_info["chunk"].push(chunk);
 
+  });
 
+  // download chunks if bedfile is an url
+  if (isURL) {
+    await retrieveChunks(bedFile, bed_info["chunk"]);
+  }
 
+  // check for a tracks.json file to prefill tracks configuration
+  for (let i = 0; i < bed_info["chunk"].length; i++) {
     let tracks = {};
     let trackID = 1;
 
-    const chunk_path = `${dataPath}${chunk}`;
-    const track_json = path.join(chunk_path, "tracks.json");
+    const dirName = bed_info["chunk"][i];
+    const chunk_path = `${dataPath}${dirName}`;
+    let track_json = `${chunk_path}/tracks.json`
```

Comment:
I think this is the third place we manually paste together stuff from the BED line to get the path to a file. The third time you just implement the same thing is about when it's time to make it a function. We probably want a function we can call to get us the local path for a file, to either put it at or read it from.

Reply:
I feel like `getChunkPath` *wants* to be the function that, given a BED URL and a chunk path string from that BED, determines the local path at which to look for stuff. But we can't use it here because it also does other stuff like actually downloading the files. Maybe it needs to get split up?